### PR TITLE
Fix #473 - Add rebuild container to mesos/zookeeper/cadvisor/dnsmasq

### LINKED ISSUE #473: Add rebuild_container to every container
Follow same approach than https://github.com/Capgemini/Apollo/pull/472 for the other containers i.e mesos master, mesos slave, dnsmasq, zookeeper.

We could also add a global variable "rebuild_all_containers" so then per container we could:
when: marathon_rebuild_container OR rebuild_all_containers

### PR DESCRIPTION
**Pull request by tayzlor**

### CODE REVIEW

File: roles/cadvisor/tasks/main.yml [hunk 1]
```diff
@@ -1,4 +1,13 @@
 # tasks for running cadvisor
+- name: destroy old cadvisor container
+  when: cadvisor_rebuild_container
```
Comment (enxebre) [inline roles/cadvisor/tasks/main.yml]: Not only for this PR we should set a convention about using "| bool" filter or not with booleans as we have both cases now. The filter is safer in terms of a value passing though via automation e.g export variable used by bash script used by ansible

Comment (tayzlor) [inline roles/cadvisor/tasks/main.yml]: so we're saying we prefer this in all instances - 

```
when: cadvisor_rebuild_container|bool
```

File: roles/marathon/tasks/main.yml [hunk 2]
```diff
@@ -9,7 +8,7 @@
     - marathon
 
 - name: create marathon artifact store directory
-  when: marathon_artifact_store_dir is defined and marathon_enabled | bool
+  when: marathon_artifact_store_dir is defined
```
Comment (wallies) [inline roles/marathon/tasks/main.yml]: do this marathon_artifact_store_dir is defined and marathon_artifact_store_dir != ''

Comment (tayzlor) [inline roles/marathon/tasks/main.yml]: Yeah there's a number of other places we have this though - do we want to create a separate issue for that? e.g. - 

```

/workspace/apollo/roles/consul/tasks/main.yml:
   57:   when: consul_join is defined
   71:   when: consul_join is defined

/workspace/apollo/roles/docker/tasks/main.yml:
   37:   when: http_proxy is defined and http_proxy != ''

/workspace/apollo/roles/marathon/tasks/main.yml:
   11:   when: marathon_artifact_store_dir is defined

/workspace/apollo/roles/mesos/defaults/main.yml:
   35: {% if mesos_master_quorum is defined %}

/workspace/apollo/roles/serverspec/tasks/main.yml:
   48:   when: test_role is defined and serverspec_run_tests

/workspace/apollo/roles/zookeeper/defaults/main.yml:
   10: 	{%- if zookeeper_host_list is defined -%}

/workspace/apollo/roles/zookeeper/templates/zoo.cfg.j2:
    6: {% if zookeeper_conf is defined %}
```

**Comment by enxebre** [comment]
lgtm, just for the record: this removes ability to stop a "service" e.g marathon or cadvisor on second deploy.

File: roles/marathon/tasks/main.yml (hunk 2 above)
Comment (wallies) [inline roles/marathon/tasks/main.yml]: yep raised #536

Comment (enxebre) [inline roles/marathon/tasks/main.yml]: then should just be 'marathon_artifact_store_dir != ''"" is defined is probably not need as they will be always defined in default.yml ?

Comment (tayzlor) [inline roles/marathon/tasks/main.yml]: maybe, lets discuss in #536

**Pull request merged**